checkbox-ng/checkbox_ng/launcher/agent.py

```diff
@@ -91,76 +97,105 @@ class RemoteAgent():
-                SessionAssistantAgent.session_assistant.resume_by_id(None)
-        else:
-            _logger.info("RemoteDebRestartStrategy")
-            if remote_restart_strategy_debug:
-                strategy = RemoteDebRestartStrategy(debug=True)
-            else:
-                strategy = RemoteDebRestartStrategy()
-            if os.path.exists(strategy.session_resume_filename):
-                with open(strategy.session_resume_filename, 'rt') as f:
-                    session_id = f.readline()
-                _logger.info(
-                    "RemoteDebRestartStrategy resume_by_id %r", session_id)
+            lambda x: None
+        )
+
+        # the agent is meant to be run only as a service,
+        # and we always resume if the session was automated,
+        # so we don't need to encode check whether we should resume
+
+        sessions = list(ctx.sa.get_resumable_sessions())
+        if sessions:
+            # the sessions are ordered by time, so the first one is the most
+            # recent one
+            if is_the_session_noninteractive(sessions[0]):
                 SessionAssistantAgent.session_assistant.resume_by_id(
-                    session_id)
+                    sessions[0].id
+                )
+
         self._server = ThreadedServer(
             SessionAssistantAgent,
             port=agent_port,
             protocol_config={
                 "allow_all_attrs": True,
                 "allow_setattr": True,
                 "sync_request_timeout": 1,
-                "propagate_SystemExit_locally": True
+                "propagate_SystemExit_locally": True,
             },
         )
         SessionAssistantAgent.session_assistant.terminate_cb = (
-            self._server.close)
+            self._server.close
+        )
         self._server.start()
 
     def register_arguments(self, parser):
-        parser.add_argument('--resume', action='store_true', help=_(
-            "resume last session"))
-        parser.add_argument('--port', type=int, default=18871, help=_(
-            "port to listen on"))
+        parser.add_argument(
+            "--resume", action="store_true", help=_("resume last session")
+        )
+        parser.add_argument(
+            "--port", type=int, default=18871, help=_("port to listen on")
+        )
+
+
+def is_the_session_noninteractive(
+    resumable_session: "ResumeCandidate",
+) -> bool:
+    """
+    Check if given session is non-interactive.
+
+    To determine that we need to take the original launcher that had been used
+    when the session was started, recreate it as a proper Launcher object, and
+    check if it's in fact non-interactive.
+    """
+    # app blob is a bytes string with a utf-8 encoded json
+    # let's decode it and parse it as json
+    app_blob = json.loads(resumable_session.metadata.app_blob.decode("utf-8"))
+    launcher = Configuration.from_text(app_blob["launcher"], "resumed session")
```

Comment:
You may want to check for the existence of the `launcher` key here. If a session is started manually without using a launcher, the session will not contain a launcher.

I just tested this by:

1. running checkbox-cli and starting a test run, then save & quitting
2. then:

```python
>>> sa = SessionAssistant()
>>> sessions = list(sa.get_resumable_sessions())
>>> app_blob = json.loads(sessions[0].metadata.app_blob.decode("utf-8"))
>>> app_blob
{'testplan_id': 'com.canonical.certification::client-cert-desktop-22-04-manual',
 'description': ''}

>>> Configuration.from_text(app_blob["launcher"], "resumed session")
KeyError: 'launcher'

```

Reply:
Fixed and added tests for the case in 12583f244